ipyleaflet/leaflet.py

```diff
@@ -497,6 +497,11 @@ class Popup(UILayer):
     auto_close = Bool(True).tag(sync=True, o=True)
     close_on_escape_key = Bool(True).tag(sync=True, o=True)
 
+    def open_popup(self, location=def_loc):
+        self.send({'msg': 'open', 'location': location})
```

Comment:
There seems to be a duplication of information between the `location` property and the `location` argument of this method.

Should this method default to the location property?
```python
def open_popup(self, location=None):
    self.send({'msg': 'open', 'location': location is None ? self.location : location})
```

And should we change the location property if it's provided?
```python
def open_popup(self, location=None):
    if location is not None:
        self.location = location
    self.send({'msg': 'open', 'location': location is None ? self.location : location})
```

Reply:
I realize I've been writing non-valid Python code in my comment ahah. Let's try again:

```python
def open_popup(self, location=None):
    self.send({'msg': 'open', 'location': self.location if location is None else location})
```

And should we change the location property if it's provided?
```python
def open_popup(self, location=None):
    if location is not None:
        self.location = location
    self.send({'msg': 'open', 'location': self.location if location is None else location})
```